crates/hir-ty/src/chalk_db.rs

```diff
@@ -152,7 +152,8 @@ impl<'a> chalk_solve::RustIrDatabase<Interner> for ChalkContext<'a> {
                 impl_maps.into_iter().chain(block_impls).for_each(&mut f);
                 def_blocks
                     .into_iter()
-                    .filter_map(|it| self.db.trait_impls_in_block(it?))
+                    .filter_map(|it| it)
```

Comment:
I can never remember if the preferred idiom is `.filter_map(|it| it)` or `.flat_map(|it| it)` or something else.

Reply:
`.flatten()` should work here as well I think?

Reply:
Changed to `.flatten()`.